include/hw_unique_key.rst

```diff
@@ -0,0 +1,90 @@
+.. _lib_hw_unique_key:
+
+Hardware unique key
+###################
+
+.. contents::
+   :local:
+   :depth: 2
+
+The hardware unique key library manages symmetric root keys on devices with the Arm CryptoCell security subsystem.
+
+Functionality
+*************
+
+Hardware unique keys (HUKs) are device-specific keys that can be used with functions for key derivation to generate other keys.
+The generated keys can then be used by applications for symmetric encryption or authentication.
+By using HUKs, you can let your application use multiple keys without having to store them on the device, as they can be recreated from a HUK when needed.
+
+.. note::
+   You should use hardware unique keys only for key derivation, but never directly for symmetric encryption or authentication.
+
+The library supports up to 3 different HUKs:
+
+* Device root key (KDR).
+  It can be used for deriving general-purpose keys.
+  On each boot it must be written to a specific KDR register in the Arm CryptoCell CC310.
+  This is done via :c:func:`hw_unique_key_load_kdr`.
+  The nRF Secure Immutable Bootloader does this automatically if :option:`CONFIG_HW_UNIQUE_KEY` is enabled in the bootloader.
+* Master key encryption key (MKEK).
+  It is used for deriving Key Encryption Keys (KEKs), which are used to encrypt other keys when these are stored.
+  It is provided to CryptoCell when it is used.
+* Master key for encrypting external storage (MEXT).
+  It is used to derive keys for encrypting data in external non-secure storage.
+  It is provided to CryptoCell when it is used.
+
+See the following table for an overview of the key types supported by each device:
+
+.. list-table::
+    :header-rows: 1
+
+    * - Device
+      - CryptoCell version
+      - Key Management Unit
+      - Supported HUK types
+    * - nRF9160
+      - CC310
+      - Yes
+      - KDR, MKEK, MEXT
+    * - nRF5340
+      - CC312
+      - Yes
+      - MKEK, MEXT
```

Comment:
53 needs to support KDR as well (destination address is the same as for MKEK and MEXT)